src/Momento.Sdk/CacheClient.cs

```diff
@@ -976,6 +976,23 @@ public async Task<CacheSetFetchResponse> SetFetchAsync(string cacheName, string
 
         return await this.DataClient.SetFetchAsync(cacheName, setName);
     }
+    
+    /// <inheritdoc />
+    public async Task<CacheSetSampleResponse> SetSampleAsync(string cacheName, string setName, int limit)
+    {
+        try
+        {
+            Utils.ArgumentNotNull(cacheName, nameof(cacheName));
+            Utils.ArgumentNotNull(setName, nameof(setName));
+            Utils.ArgumentNotNull(limit, nameof(limit));
```

Comment:
I think you meant to check if the limit was non-negative here. We have an `ArgumentStrictlyPositive` helper in the `Utils` namespace; we can make a new one `ArgumentNonnegative` companion as opposed to testing vs null.